prog/postgres/converge_postgres_resource.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require "forwardable"
+
+require_relative "../../lib/util"
+
+class Prog::Postgres::ConvergePostgresResource < Prog::Base
+  subject_is :postgres_resource
+
+  label def provision_servers
+    hop_wait_servers_to_be_ready if postgres_resource.has_enough_fresh_servers?
+
+    if postgres_resource.servers.none? { _1.vm.vm_host.nil? }
+      exclude_host_ids = (Config.development? || Config.is_e2e) ? [] : postgres_resource.servers.map { _1.vm.vm_host.id }
+      Prog::Postgres::PostgresServerNexus.assemble(resource_id: postgres_resource.id, timeline_id: postgres_resource.timeline.id, timeline_access: "fetch", exclude_host_ids: exclude_host_ids)
+    end
+
+    nap 5
+  end
+
+  label def wait_servers_to_be_ready
+    hop_provision_servers unless postgres_resource.has_enough_fresh_servers?
```

Comment:
If I'm not mistaken, in an absurd case of multiple back-to-back repeated updates on the target, this code would jump back and forward to provision many servers, and the excess servers would be pruned at the end only. This can cause an capacity alert. This is a very hypothetical scenario and easy to mitigate once a page occurs, but just wanted to note if the logic makes sense. 

Maybe in order to avoid this and some unforeseen concurrency issues, we can disable patching while an update is going on in the front end PR.

Reply:
Yeah, it is a possibility, but a quite rare one. I prefer to keep the logic simple for now and allow users to change target 1-2 times if they change their mind. We can change the logic if people start to abuse this.